tests/always/test_project_structure.py

```diff
@@ -90,8 +89,10 @@ def test_providers_modules_should_have_tests(self):
 
         missing_tests_files = expected_test_files - expected_test_files.intersection(current_test_files)
 
-        with self.subTest("Detect missing tests in providers module"):
+        try:
```

Comment:
Hmm. That looks strange. I am not sure if I understand why we have try/Except at all (assert with description should be enough) and I am not sure what was the effect of self.subTest here. Can you please explain @Abhishek-kumar-samsung ?

Reply:
I explained just below.

Reply:
I see. Makes sense for subtest removal. 

But why not just this:

```
assert set() == missing_tests_files, "Detect missing tests in providers module"
```

Reply:
I didn't knew if it can be done that way, i knew try catch way, and tried testing by putting missing_tests_files as none, try except way was working fine so i kept that.
And also as per understanding, try except was much easy to understand.

Reply:
if you want, i can try your way @potiuk.

Reply:
It's just unnecessary. What assert does it throws an exception, fail is just a shorthand for "assert false"  so you are effectively running assert, catch exception, and run another assert. The way I proposed is the pytest-y way of doing what you want to do.

Reply:
oh okay, i will modify it.

Reply:
And no. For me in this case try/except is actually confusing, because it makes the person reading it to think "is there any reason why this is done this way? Did I miss something?".

Reply:
@potiuk done as you have said.

Reply:
Well.. :)